zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/DataStoreManager.scala

```diff
@@ -161,12 +161,33 @@ class DataStoreManager(metaDataset: String,
     val dropTableName = dropTable.dataset
 
     if(metaData.contains(dropTableName)){
-      metaData.remove(dropTableName)
+      context.child("data-" + dropTableName) match {
+        case Some(datasetActor) => datasetActor ! PoisonPill
+        case None =>
+      }
 
-      //TODO send query to DELETE dataset in database
+      val metaRecordFilter = FilterStatement(DataSetInfo.MetaSchema.fieldMap("name"), None, Relation.matches, Seq(dropTableName))
+      metaActor ! DeleteRecord(metaDataset, Seq(metaRecordFilter))
 
-      sender ! DataManagerResponse(true, "Deregister Finished: dataset " + dropTableName + " has successfully removed.\n")
+      metaData.foreach( dataSchema =>
+        dataSchema._2.createQueryOpt match {
+          case Some(query) =>
```

Comment:
Got it. I searched this and it seems that if it is a `Unit` function in `map`, `foreach` is better.